elasticdl/python/worker/worker.py

```diff
@@ -547,31 +548,48 @@ def forward_process(self, features):
         outputs = self._model.call(features, training=False)
         return outputs
 
-    # TODO: Reuse model handler to initialize the model parameters properly
     def _update_local_model_params(self, model_params):
-        pass
+        self._non_embed_vars = model_params
 
     def _get_local_model_params(self):
-        return {}
+        return self._non_embed_vars
 
     # TODO: Implement master gRPC service to select a worker
     # to be used for broadcast model parameters from.
-    def _get_broadcast_root_worker_ip(self):
+    @staticmethod
+    def _get_ip_of_broadcast_root_worker():
         return 1
 
+    @staticmethod
+    def _get_ip_of_this_worker():
+        hostname = socket.gethostname()
+        # Use localhost for testing locally on a laptop or inside a container
+        if ".local" in hostname or "." not in hostname:
+            hostname = "localhost"
+        return socket.gethostbyname(hostname)
+
     def _broadcast_model_params(self):
         status = self._collective_communicator.barrier()
         if status == CollectiveCommunicatorStatus.FAILED:
             logger.warning("Failed to perform barrier operation")
             return False
-        status, model_params = self._collective_communicator.broadcast(
-            self._get_local_model_params(),
-            self._get_broadcast_root_worker_ip(),
+        broadcast_root_worker_ip = self._get_ip_of_broadcast_root_worker()
+        this_worker_ip = self._get_ip_of_this_worker()
+        is_broadcast_root_worker = this_worker_ip == broadcast_root_worker_ip
+        model_params = (
+            self._get_local_model_params()
+            if is_broadcast_root_worker
+            else None
         )
-        if status == CollectiveCommunicatorStatus.FAILED:
-            logger.warning("Failed to broadcast model parameters")
-            return False
-        self._update_local_model_params(model_params)
+
+        if not is_broadcast_root_worker and model_params is not None:
```

Comment:
If the current worker is_broadcast_root_worker,  it also needs to broadcast its model_params to other workers.

Reply:
Good catch! This should be fixed now.